### ISSUE TITLE
Downloading and Caching HLS streams like m3u8 in app 

### ISSUE THREAD

**Issue by seanjackson** [Baseflow/flutter_cache_manager#386]
I am developing an application like netflix and youtube where users can download the videos in app.I want to use the flutter cache manager.Please can someone guide me on how to downloads hls streams with flutter cache manager. I am on a serious deadline
Thanks

**Comment by lianfanghua** [comment]
May be you can download m3u8 file first, and then use [flutter_hls_parser](https://pub.dev/packages/flutter_hls_parser) to parse m3u8 file, and get the media url from m3u8 file. after that, you can download the media file yourself.

Or you can use [better_player](https://pub.dev/packages/better_player), it provides caching solution(but you can't manage it yourself).

**Comment by codesniper9315** [comment]
Hi, @lianfanghua The **better_player** supports cache option while playing the HLS video?

**Comment by lianfanghua** [comment]
> Hi, @lianfanghua The **better_player** supports cache option while playing the HLS video?

Yes, you can see better_player's [office documentation](https://jhomlala.github.io/betterplayer/#/cacheconfiguration) about caching.
At the tail of the document, there is an explanation of the caching scheme of HLS

**Comment by lianfanghua** [comment]
But better_player only provides a caching solution, not an Offline solution, if you want users to be able to download HLS media manually, you still have to provide it yourself.

**Comment by seanjackson** [comment]
@lianfanghua  Thanks for everything.You are the best! do you know how to implement hls download like Netflix or YouTube whereby users can watch the movies in app.

**Comment by lianfanghua** [comment]
> @lianfanghua Thanks for everything.You are the best! do you know how to implement hls download like Netflix or YouTube whereby users can watch the movies in app.

I don't think it has anything to do with HLS's caching solution or offline solution, I have no experience with this.

**Comment by jynd** [comment]
HLS doesn't support play cached video, because HLS is http live streaming. You can use ffmpeg to convert hls to mp4 and save it in the storage. Then, when the user open app in the next time, get mp4 file in the storage and play it. every activities convert hls to mp4 might run in isolate, avoid block UI

**Comment by roshTailor** [comment]
> May be you can download m3u8 file first, and then use [flutter_hls_parser](https://pub.dev/packages/flutter_hls_parser) to parse m3u8 file, and get the media url from m3u8 file. after that, you can download the media file yourself.
> 
> Or you can use [better_player](https://pub.dev/packages/better_player), it provides caching solution(but you can't manage it yourself).

how to download m3u8 file firstly
